### ISSUE TITLE
MD022 config and MD012 are conflicting

### ISSUE THREAD

**Issue by robsdedude** [rvben/rumdl#429]
### rumdl version

0.1.22

### Installation method

Pre-built binary

### Bug description

Configuring MD022 with more than one line above or below the headline conflicts with  `MD012`.

### Steps to reproduce

Assuming the following config

```toml
[MD022]
lines-above = 2
```

It's impossible to make the linter happy.

```markdown
# Title


## Subtitle
```
yields
```
test.md:3:1: [MD012] Multiple consecutive blank lines between content [*]
```

```markdown
# Title

## Subtitle
```
yields
```markdown
test.md:3:1: [MD022] Expected 2 blank lines above heading [*]
```

### Expected behavior

Have the first version of `test.md` pass without lints (and without disabling `MD012`, which is still useful generally, just not around headlines).

### Actual behavior

Getting a lint issued no matter what.

### Operating system

Linux

### Additional context

I suggest to just skip applying rule MD012 if the content before or after the blank lines is a headline. It should be covered by rule MD022 already either way (given MD022 is enabled).

**Comment by rvben** [comment]
Hi @robsdedude, this is an interesting one. While we match markdownlint behavior in this one, there's no reason why we shouldn't be able to handle this better.

**Comment by rvben** [comment]
Fixed in v0.1.23. MD012 now skips blank line sequences adjacent to ATX and Setext headings, so MD022's `lines-above`/`lines-below` settings can coexist without triggering spurious MD012 violations.
